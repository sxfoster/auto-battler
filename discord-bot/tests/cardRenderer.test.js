const fs = require('fs/promises');
const path = require('path');
const sharp = require('sharp');
const { generateCardImage } = require('../src/utils/cardRenderer');

const assetsDir = path.join(__dirname, '..', 'assets');
const framesDir = path.join(assetsDir, 'frames');

beforeAll(async () => {
  await fs.mkdir(framesDir, { recursive: true });
  const frameBuffer = await sharp({
    create: {
      width: 300,
      height: 400,
      channels: 4,
      background: { r: 30, g: 60, b: 90, alpha: 1 }
    }
  }).png().toBuffer();
  await fs.writeFile(path.join(framesDir, 'frame-common.png'), frameBuffer);
});

afterAll(async () => {
  await fs.rm(framesDir, { recursive: true, force: true });
});

describe('generateCardImage', () => {
  test('produces consistent card image', async () => {
    const card = {
      name: 'Test Hero',
      class: 'Warrior',
      rarity: 'Common',
      hp: 10,
      attack: 5,
      speed: 3
    };
    const buffer = await generateCardImage(card);
    expect(buffer.toString('base64')).toMatchSnapshot();
  });
});
