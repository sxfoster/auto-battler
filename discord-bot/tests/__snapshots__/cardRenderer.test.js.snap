// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`generateCardImage produces consistent card image 1`] = `"iVBORw0KGgoAAAANSUhEUgAAASwAAAGQCAYAAAAUdV17AAAACXBIWXMAAAPoAAAD6AG1e1JrAAAVFUlEQVR4nO3ch7cTdd7H8f1Lsr0+W8+259lVsWFf26q7UgUEQVAQkCYIIiBIURFQUBQLgmBDQQUVLIgFFVFWEQVERUAREUE6n+d855xcUyb3Jpd7TT7se855HSCZJJMy78z8ZsL3Mke1EgBkDHyv2gsAAJkyESwAckGwAMgFwQIgFwQLgFwQLAByQbAAyAXBAiAXBAuAXBAsAHJBsADIBcECIBcEC4BcECwAckGwAMgFwQIgFwQLgFwQLAByQbAAyAXBAiAXBAuAXBAsAHJBsADIBcECIBcEC4BcECwAckGwAMgFwQIgFwQLgFwQLAByQbAAyAXBAiAXBAuAXBAsAHJBsADIBcECIBcEC4BcECwAckGwAMgFwQIgFwQLgFwQLAByQbAAyAXBAiAXBAuAXBAsAHJBsADIBcECIBcEC4BcECwAckGwAMgFwQIgFwQLgFwQLAByQbAAyAXBAiAXBAuAXBAsAHJBsADIBcECIBcEC4BcECwAckGwAMgFwQIgFwQLgFwQLAByQbAAyAXBAiAXBAuAXBAsAHJBsADIBcECIBcEC4BcECwAckGwAMgFwQIgFwQLgFwQLAByQbAAyAXBAiAXBAuAXBAsAHJBsADIBcECIBcEC4BcECwAckGwAMgFwQIgFwQLgFwQLAByQbAAyAXBAiAXBAuAXBAsAHJBsADIBcECIBcEC4BcECwAckGwAMgFwQIgFwQLgFwQLAByQbAAyAXBAiAXBAuAXBAsAHJBsADIBcECIBcEC4BcECwAckGwAMgFwQIgFwQLgFwQLAByQbAAyAXBAiAXBAuAXBAsAHJBsADIBcECIBcEC4BcECwAckGwAMgFwQIgFwQLgFwQLAByQbAAyAXBAiAXBAuAXBAsAHJBsADIBcECIBcEC4BcECwAckGwAMgFwQIgFwQLgFwQLAByQbAAyAXBAiAXBAuAXBAsAHJBsADIBcECIBcEC4BcECwAckGwAMgFwQIgFwQLgFwQLAByQbAAyAXBAiAXBAuAXBAsAHJBsADIBcECIBcEC4BcECwAckGwAMgFwQIgFwQLgFwQLAByQbAAyAXBAiAXBAuAXBAsAHJBsADIBcECIBcEC4BcECwAckGwAMgFwQIgFwQLgFwQLAByQbAAyAXBAiAXBAuAXBAsAHJBsADIBcECIBcEC4BcECwAckGwAMgFwQIgFwQLgFwQLAByQbAAyAXBAiAXBAuAXBAsADYIFgC5IFgA5IJgAZALggVALggWALkgWADkgmABkAuCBUAuCBYAuSBYAOSCYAGQC4IFQC4IFgC5IFgA5IJgAZALggVALggWALkgWADkgmABkAuCBUAuCBYAuSBYAOSCYAGQC4IFQC4IFgC5IFgA5IJgAZALggVALggWALkgWADkgmABkAuCBUAuCBYAuSBYAOSCYAGQC4IFQC4IFgC5IFgA5IJgAZALggVALggWALkgWADkgmABkAuCBUAuCBYAuSBYAOSCYAGQC4IFQC4IFgC5IFgA5IJgAZALggVALggWALkgWADkgmABkAuCBUAuCBYAuSBYAOSCYAGQC4IFQC4IFgC5IFgA5IJgAZALggVALggWALkgWADkgmABkAuCBUAuCBYAuSBYAOSCYAGQC4IFQC4IFgC5IFgA5IJgAZALggVALggWALkgWADkgmABkAuCBUAuCBYAuSBYAOSCYAGQC4IFQC4IFgC5IFgA5IJgAZALggVALggWALkgWADkgmABkAuCBUAuCBYAuSBYAOSCYAGQC4IFQC4IFgC5IFgA5IJgAZALggVALggWALkgWADkgmABkAuCBUAuCBYAuSBYAOSCYAGQC4IFQC4IFgC5IFgA5IJgAZALggVALggWALkgWADkgmABkAuCBUAuCBYAuSBYAOSCYAGQC4IFQC4IFgC5IFgA5IJgAZALggVALggWALkgWADkgmABkAuCBUAuCBYAuSBYAOSCYAGQC4LVTK67ZdZh+dVJnar+HA7XCW2uLHpevz2tS+q8fzija9G8x7XqU/XngFY1hWA1g+8f3VqHO/3vuZdVZdnXbvhUGzZuqXNS+/6Nvq/Lr5lU9LyOb903dd6T2w8omrfbkJuq/l6iVU0hWM3AOVj79u/PW44zLh7c6PsiWMg0MYLVTMHavmNnkR07dxWtlDt37U6d9y/n9KjKshMsZGoYwfoOndZxUNFK2bH/uHpv8/MTO+jCniPVZ9StGjh2unoMu1ktLuxd1uP98cxuuqDHtUk44rY9h09Wuz5j9Pfze+bN94NjWuuoC3ol9u8/kLd8Fw8cX3dd+M2pnWsmWMde2EeXXj0xeW59R01Vq16j9LMTLio5/4+PbZv3XEL2uj+ddam6D52oQeOml4x0zN918I3J4/Ufc5s69BubjL1V+3OV+S9CsGo0WL87vYvufeRp7d6zV2nTynfX6rzuw1NvGyvfohde08GDB1Vqiq24kZNnJvPHQHi505R7H616sOLy99Z9nLp8u77Zrcn3zNNPj29fdLsYjyuc4gvhttkLdODAt69V/Dv3dhGmVWvWpz7eoUOH9MyyFWrZrvFjfWhVNoJVg8GKb/KNm7c2GI/YGooo5N72Ry3aavXaj1TOdM/DT9kFa+qs+WUt52tvr0li1FCw4jUonG6///G621w/7f4kSg1N3+zeo7Z9Rlf9M5Y5whGsGgtWBKdw62HTZ19ozNTZ6jd6muY9tSzvutgCi9MHsreP3aLcKY70jZpyn3pdO0XXTLxH0+c8oRX/+SBZCbPB+vUpF9cdFSxcOeOxc48axn0dTrDiub21el2RtC2mwmD1Hnlr3vWxrA8vWprsno2+dba2bP0y7/pZjy1uMFi50569+5I/4zWK+dv3HVP0esRjPLLoRT353PJkay53ijHK2Lqt9ucscwQjWDUWrMKVcuu2r5KxqNx5Zjy4MG+ehxYurbsuVt7cKcZl0pYljkKe1eXq73zQvZIpN1gxzvbplvytzrHT5hRtmebuQsduXu54XVqwYrc5YpcNTfx5eqerkr/HbnfutGb9J3ljVqdcNLAoWoW7k2jVpAhWjQUrvrlzp2mzileACE3heFSs0HFdrOS5U2y5xEDyPy+9JtmSamgZazVYsRy50959+1NPrl362qq8+YZMuLPeYM2evyR12SNchVNspRbOd/dD+buU6z/eVPXPWeYIRrBqLFjvf7ixUSv3n8/untz+92dckoynlJo+/GRzsksTA8nVCNZjz7ykmfOeKbJgySv1Biti0ZgptkbrC9aZXYakLvu/LhtRNG/a0dkrRtySN0/sQsZufbU/a5kjFMGqsWBt/nxbo1bM3HGsS666IfWcr8JpzoJnbc7Dii2lxkwxxlVfsOJobNrydBowvmjetJ8VXXTl9WXNh1ZNgmDVWLAKB59fXbk6dYukUHYLKyvOl4pzk+K6OCRfaqvrH50HWwQrziHLnWLsqJzXJQ5U1BesODcrbXnO7zG8rGWP17hwTOyHLdpU/bOWOUIRrBoL1qNP5x8FjJWuqR7/xLb9kiNyudPQG+/Km6cwbGdfUjwwX41gFcYmdr0qPWmzkmDFrnXheWwDrr+9aL65jz+XN0984VT7c5Y5ghGsGgtW4aB5HGo/t9uw1PuLI2BxkuT42+fWXRbjLDHIXmq35L5HF+fdf5zqUPjj58Lzrn7RMv98pmqdh7Xuo01F42Fp40U/Oa6dugy6Qa+vWpP8vTHBCi+veCdv3jitI/e0hRjnyp4KkZ0mzni46p+zzBGMYNVYsOJ3iG+sej9vnvimX/j8ct0046EkTnEGfOzmZc8RipMps7ePuGWPor2w/O3knKI4+XHcbXM0f/HLRecVxU93ch//qaWvq76pkjO6mzpYnQdOKFr+iMgdc59MTk2IuMbyx1HT7HQ4wYojq7lnwGd/+xlHIiOGhT9j+uyL7YxfHdW8CFYNnuke3+LvfrBB5U5pwSpnWvLSm0WP3ab3dfWe2V3NYIURk2YWRaS+6XCClT2vrTBMadO27TtSz2tDqyZFsGr0t4TxW7jYMio8WTI7xUobW2ITpj+Q/Ag4e7u/nXd5copAfT/t+Xzbdt181yOpv7cL8ROTiFlsvRQOwlc7WOGcrkO1+KUVJUMSz++BJ57XZcPyf7bUmGCFODARj5cWyjhRNc7l+uu51fnfNTL/ZQiWgRirip/cxAocu3DxP3EW/k4uTQxKn9phYBKguG38GXHLnmTqLl6DiFecUxbiDPXYOo3d6uZ4vF+27JjsJsb/YBGPFyErJ3ho1WQIFgC5IFgA5IJgAZALggVALggWALkgWADkgmABkAuCBUAuCBYAuSBYAOSCYAGQC4IFQC4IFgC5IFgA5IJgAZALggVALggWALkgWADkgmABkAuCBUAuCBYAuSBYAOSCYAGQC4IFQC4IFgC5IFgA5IJgAZALggVALghWgYsHjtf8xS/rt6d1Kbruz2d3T65r22d03WXt+45JLsuas+BZjZ02Rye0ubJRj//DFm10brdhGnbT3Zr7+HPJfR7Xqk/J+XtdO0Uvr3hHazd8qqeWvq6zL7m6osf7UYu2emjhUj369DL9/MQOdZdPuvuRvOdVyv+c3En/vnxE8ve/nXd53n3/7ISLdNeDi/TAE8/rj2d2q2i5Wrbrn/p4s+cvKev2vz/jEk2+Z17da7P8rfeS23YeOCFvvtz7jtdg6qz56jHs5uR9yM4T70fufA8++YKmz3lCvUfemvo5qcR53Ydr1mOL9fqqNVr30Sa9unK1Js54WL87/fDuN3OEIlgFrrtllmL6yzk9iq6LcMQ0ZMKddZdFWGKa99QyzZz3jB5/9hVt37FT+/bvL1o5yhErfUx79+3Xxs1bk7/HCpM274hJM5Pr4wM/aNx0LXvjP8ntzuk6tKJAx3To0KFkBSwVrE82f548p7Rg9Rs9LbmPUy4aWHf7uPyVN9/V17u+0b8uG1Hx69Cq16jkPp97ZWXFwYr37rMvtmvb9h26bfYCDb3xLk2599Fked79YEPevDG9t+7j5L2LL4hVa9bXPe4PjmmdzHPZsEnJZQufX57M9/CipXrx9VX6Zvce7dm7L/mCauznLaIa71ss5+hbZyfPL+73/Q836lcndar6+pCpMQSriYJ19L+uqLvsT2ddqs+3bU++2St9/NjKOb3TVfrxsW3VdfCNJYP1m1M7a9c3u3X//GfrLovbxGPGt3W5jxdbZW+tXpesNLE1Umq+WEl37NyVel1hsGLrIO4zgvGPzoMb9T5kg3Vi234V3zYCElPaVu4fzuia9++Y7nxgYd5ld8x9Mrm8XZ8xecEq/CL49SkXJ69/TPG5aarPYJdBNyT3efk1k5r1s54xRLCaIVghdoNiqyV2i+Lffz+/Z7IS5u52NaS+YMVuS0yFWy8Tpj9QcvkLxW7a/v0HdPUNM5Ktq1jewudRabAi1rHFsvnzbamxOb513+R1yN3laupgxZZSxLycedOCdVrHQXkRKhWs8P2jW+uNVe8nW5K/aNkh7z7O7zG8UZ/B+KzEFFuG3/XnP1PjCFYzBSu2XHbv2Vu3WzH+9rklv/UbE6zrp92fXPfLlh3zLo8xtZhibKSh+x85eWaymxfjPXE/sZLH+EljgxVbBhs2bkkcdUGv1HljlyqmhnZ3ssGKcaXYFYutnm5Dbqp7Pct5DyPeha9POcHKPvZV4+9oMFhh4NjpyfWtr7iu7rIXlr+tL77cUfHnL6IXr9HOXbv1f//MHxPMgGCV+rDHmE125cvKjinVF6zYguo7aqoOHjyYjINk5+s5fHLyIa7kQ1hfsGY8uDAZryq8/MwuQ5LbdB86scH7X7P+kyQGuVuF8Rxjq6ExwYqxl0+3bE0OTpR6zHi94nXIbnmWEtGI3eoI/yOLXkzGdGJ6ZtmK5EBBfbeN9yAGr2OKLcjYPY3B9LQDErnBiucdW3Qxf7x/2a27hoJ1QY9rk+v7j7mt7rIYM1uw5JWy3+vYjd/02RfJ8sZzjYMO1V4XMjWIYJUIVgw6x99zxYe+VLAKpzff+eCwvyHrC1YMtKft9pzaYWBym9wB9DSx8ma3irKXXdhzZNGWQiXByq7o8e/DfR8iaLlbUxGT2PrL3fKpT8wfR3PjKGUMpMdyxRRH93LnS5ti62bA9bfXzdNQsOL9iWnw+G8/F5WKL7T4XMXyffX1Lr329ppknLLa60OmxhCsJtoljHGg2GWJlb7U7lBTBiu78v7kuHapuzNtehdHJ1fsdsS3eQzsxnKHS6+emGwlxRHPxgQrDhbEVkXE4crrDj9aaRGKI7Cx1VXpbWOr7/lX30qWM07DyF4e05KX3kyef6cB45PgF44zNhSseN1iiverKZ5nbF0dOHAw2Uprzs96xhDBaqYxrOYOVpzGEFPhOVqxZdDQYHVsvUR8Sk0x9lb47V7uoHvsrsWpHRGt2DVu6tckzlV66Y3SRzPrc3L7Aclyjpk6O3WXsJSGgpU9wFLOgY5yrf94U7Lr3NSvX8YcwTINVgzexzRqyn15l8cYz5atX6aOQ+WebFrqfuMoXtpuVyWnNUS0nnxueRKtPqPq3zWtRIsLeydbHrF1WN98cXpH2uWxZRXTNRPvabJgdew/LlmmiHRTPc/4sogvjRi7a+rPVMYcwfqOglXJUcIO/cYmuyjTZi1IbjPutjnJv2N3L3e+2DX68quvkzGn+JDHbml80w+/+dsVMk2cc/Xxps9KXh9jUSvfXdvoYGWjEQP6Ea3c8bRyjxLeet9jSexiNzNe9ziIEAc+4kTNk9rXPyAdjxG7tfE6HvPvK5L3Mk7i/fCTzckub+4Z+ZUEK04ziPcgxv1ivCpe/3h+8XoVnvFe7lHC2DqL9za2/uJ5xm5pjH9GBGMwv9rrQ6bGEKwaDFZsIaVNMXicO19E6tmXV9ZdH6coROTqu+8YX4uoxRnWpea59uZ7k/vLPVJVabCy0Vr0wmvJSn3FiFsqClackBnPJ3eK87tyx5/qG8COiMTzzJ1Wr/2oKAKVBCt3lzmOhj794htJVNOOWpYbrIheHO3NneLk3whXtdeFTA0iWEeAv57bI9kSiTOvq70sTSnG2mIXNULYmCNmcQ5WfEFEeA/3N3/N/TxjOWMrq5aXM1MDCBYAuSBYAOSCYAGQC4IFQC4IFgC5IFgA5IJgAZALggVALggWALkgWADkgmABkAuCBUAuCBYAuSBYAOSCYAGQC4IFQC4IFgC5IFgA5OL/AcUwiYzWQqY5AAAAAElFTkSuQmCC"`;
